### ISSUE TITLE
Replace Ubuntu 18.04 with 22.04 in CI configuration

### ISSUE THREAD

**Issue by perlun** [perlang-org/perlang#387]
Ubuntu 18.04 is going EOL 2023-04-30, in just a little more than two weeks from now. Once this happens (or even slightly before it), we can start preparing for dropping the Ubuntu 18.04 runs in our CI config. At the same time (in the same or in a separate PR), we should also add Ubuntu 22.04 support.

_Update_: While writing this and waiting on #386 to complete, we might have to be even a bit faster. The Ubuntu 18.04 runners (`ubuntu-18.04`) might have been removed already; at least, no runner(s) are picking up my job: https://github.com/perlang-org/perlang/actions/runs/4703536383/jobs/8342654106. (This could potentially be a deliberate "brownout")

More details on the Ubuntu 18.04 deprecation from GitHub: https://github.blog/changelog/2022-08-09-github-actions-the-ubuntu-18-04-actions-runner-image-is-being-deprecated-and-will-be-removed-by-12-1-22/
